pages/blog/graphql-deep-dive-1.mdx

```diff
@@ -0,0 +1,354 @@
+
+And while I can talk more about this, it will never end cause the ecosystem is too huge and thriving.
+
+**I use REACT/Angular /Vue/Web Components. How do I integrate GraphQL with my components?**
+
+Again, the front end ecosystem is huge as well with its own set of tooling and libraries.
+
+In my case, I typically try to work on the frontend without any framework (I use [Lit Elements](https://lit-element.polymer-project.org/) in my case and we will have a separate blog on that soon), the tool you use completely depends on your requirements here.
+
+- Apollo Client does have a good integration with these frameworks including [React](https://www.apollographql.com/docs/react/), [iOS](https://www.apollographql.com/docs/ios/) and [**Android**](https://www.apollographql.com/docs/android/) — so, you might want to check that out
+- Using Vue? You can try [**Vue Apollo**](https://apollo.vuejs.org/guide/)
+- Using web components with Apollo Stack for GQL? You might want to check out [**Apollo Elements**](https://github.com/apollo-elements/apollo-elements)
+- Using vanilla JS or TS or using web components and want to have a framework-independent way of doing things? You can stick to the GraphQL codegen itself since it takes care of almost everything underneath. Or if you want, you can also use Apollo Client’s vanilla version `@apollo/client/core`.
+  [**Apollo Elements**](https://github.com/apollo-elements/apollo-elements) does come with support for a lot of webcomponent libraries like [Lit](https://lit-element.polymer-project.org/), [Fast](https://www.fast.design/) and [Gluon](https://github.com/ruphin/gluonjs) or even without any of it and hence is quite flexible.
+
+Well, there are lot of other ways we haven’t talked about and this is just a start.
+
+**What are some of the ways in which I can maintain performance while using GraphQL?**
+
+While GraphQL is really promising and helpful, you have to understand that like any technology or framework, it does come with its own set of problems, most of which have already been addressed. For instance you might have heard about the N+1 problem, lack of caching, Query cost and complexity and so on and these have been addressed by some projects like the [Dataloader](https://github.com/graphql/dataloader), [Persisted Queries](https://www.apollographql.com/docs/apollo-server/performance/apq/), [Caching](https://www.apollographql.com/docs/apollo-server/performance/caching/) and more which you can setup depending on your needs.
+
+Ultimately it depends on the degree of the flexibility you want to offer. The more the flexibility, the more the cost. And it is your decision to decide it based on your usecase.
+
+**What are some of the principles or standards to be followed when trying to build my datagraph architecture?**
+
+Some amazing people have already answered this [**here**](https://principledgraphql.com/) and I highly recommend you to go through it before starting off your journey with GraphQL.
```

Comment:
I would add here http://graphql-rules.com and personally I would be careful following principledgraphql.com as it is very opinionated towards a certain product....

Reply:
Sure. I will add the link as well. The main difference I see between `graphql-rules` and `principledgraphql` is that principledgraphql talks about the concepts in general and not the implementation itself, but I see `graphql-rules` talking about the implementation itself. So, I am not sure what is currently a replacement, but I will add both.